tests/math_tools/test_gradient.cpp

```diff
@@ -0,0 +1,147 @@
+     */
+    KOKKOS_DEFAULTED_FUNCTION GradientTestFunction() = default;
+
+    /**
+     * @brief Default copy constructor.
+     */
+    KOKKOS_DEFAULTED_FUNCTION GradientTestFunction(GradientTestFunction const&) = default;
+
+    /**
+     * @brief Default destructor.
+     */
+    KOKKOS_DEFAULTED_FUNCTION ~GradientTestFunction() = default;
+
+    /**
+     * @brief Get the value of the function at given coordinate.
+     *
+     * @param[in] coord_rth The coordinate where we want to evaluate
+     * the function.
+     *
+     * @return The value of the function at the coordinate.
+     */
+    KOKKOS_FUNCTION double operator()(CoordRTheta const coord_rth) const
+    {
+        double const r = ddc::get<R>(coord_rth);
+        double const theta = ddc::get<Theta>(coord_rth);
+        return ipow(r, 2) * Kokkos::cos(theta) + 2 * ipow(r, 3) * Kokkos::sin(theta);
+    }
+
+    /**
+     * @brief Get the value of the gradient of the field 
+     * at a given coordinate. The gradient is expressed in 
+     * the contravariant basis of the circular geometry. 
+     *
+     * @param[in] coord_rth The coordinate where we want to evaluate 
+     * the gradient.
+     *
+     * @return A vector type representing the value of the gradient.
+     */
+    KOKKOS_INLINE_FUNCTION DVectorType gradient(CoordRTheta const coord_rth) const
+    {
+        double const r = ddc::get<R>(coord_rth);
+        double const theta = ddc::get<Theta>(coord_rth);
+        DVectorType gradient(
+                2 * r * Kokkos::cos(theta) + 6 * ipow(r, 2) * Kokkos::sin(theta),
+                -Kokkos::sin(theta) + 2 * r * Kokkos::cos(theta));
+
+        return gradient;
+    }
+
+    /**
+     * @brief Get the value of the partial derivatives of the field 
+     * at a given coordinate.
+     *
+     * @param[in] coord_rth The coordinate where we want to evaluate 
+     * the partial derivatives.
+     *
+     * @return A vector type representing the value of the partial derivatives.
+     */
+    KOKKOS_INLINE_FUNCTION DVectorCovType partial_derivatives(CoordRTheta const coord_rth) const
+    {
+        double const r = ddc::get<R>(coord_rth);
+        double const theta = ddc::get<Theta>(coord_rth);
+        DVectorCovType derivatives(
+                2 * r * Kokkos::cos(theta) + 6 * ipow(r, 2) * Kokkos::sin(theta),
+                -ipow(r, 2) * Kokkos::sin(theta) + 2 * ipow(r, 3) * Kokkos::cos(theta));
+        return derivatives;
+    }
+};
+
+
+TEST(GradientTest, Circular)
+{
+    double const TOL = 1e-12;
+    CircularToCartesian<R, Theta, X, Y> const mapping;
+    FieldMemRTheta_host<CoordRTheta> coords = get_example_coords(IdxStepR(32), IdxStepTheta(32));
+    IdxRangeRTheta grid = get_idx_range(coords);
+
+    using MetricTensorType
+            = MetricTensorEvaluator<CircularToCartesian<R, Theta, X, Y>, CoordRTheta>;
+    MetricTensorType metric_tensor(mapping);
+    Gradient<MetricTensorType> gradient_evaluator(metric_tensor);
```

Comment:
FYI, you could also do:
```suggestion
    MetricTensorEvaluator<CircularToCartesian<R, Theta, X, Y>, CoordRTheta> metric_tensor(mapping);
    Gradient gradient_evaluator(metric_tensor);
```
The template parameter can be auto-deduced from the constructor.

But the syntax that you are using here is more explicit :+1: